.github/workflows/release.yml

```diff
@@ -0,0 +1,267 @@
+        gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
+        gpg-passphrase: MAVEN_GPG_PASSPHRASE
+
+    - name: Set up JDK 8
+      uses: AdoptOpenJDK/install-jdk@v1
+      with:
+        version: '8'
+        targets: 'JDK8_HOME'
+
+    # Using Python 3.8
+    # Might be worth to bump the version and/or make it configurable in the future
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Set up Maven toolchains.xml
+      run: |
+        cat << 'EOT' > ${HOME}/.m2/toolchains.xml
+        <?xml version="1.0" encoding="UTF8"?>
+        <toolchains>
+          <toolchain>
+            <type>jdk</type>
+            <provides>
+              <version>11</version>
+              <vendor>zulu</vendor>
+            </provides>
+            <configuration>
+              <jdkHome>${env.JAVA_HOME}</jdkHome>
+            </configuration>
+          </toolchain>
+          <toolchain>
+            <type>jdk</type>
+            <provides>
+              <version>1.8</version>
+              <vendor>adoptopenjdk</vendor>
+            </provides>
+            <configuration>
+              <jdkHome>${env.JDK8_HOME}</jdkHome>
+            </configuration>
+          </toolchain>
+        </toolchains>
+        EOT
+
+    # TODO Maybe better to _not_ use cached dependencies just to be sure that everything's available at time of release --> remove this one?
+    - name: Cache local pip repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+
+    # TODO Maybe better to _not_ use cached dependencies just to be sure that everything's available at time of release --> remove this one?
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.m2/repository
+          !~/.m2/repository/org/projectnessie
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    # TODO Maybe better to _not_ use cached dependencies just to be sure that everything's available at time of release --> remove this one?
+    - name: Cache local NPM repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    # TODO Maybe better to _not_ use cached dependencies just to be sure that everything's available at time of release --> remove this one?
+    - name: Cache local Gradle cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+        restore-keys: ${{ runner.os }}-gradle
+
+    - name: Prepare git config
```

Comment:
I feel so-so on this (and a lot of the secrets). While i dont want to go marketplace crazy coudlnt we use an action for this?